plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/StmtConversionVisitor.kt

```diff
@@ -180,26 +180,26 @@ object StmtConversionVisitor : FirVisitor<Exp, StmtConversionContext>() {
             return specialFunc.convertCall(getArgs(), data)
         }
 
-        val symbol = functionCall.calleeReference.resolved!!.resolvedSymbol
-        val calleeSig = when (symbol) {
+        val calleeSig = when (val symbol = functionCall.toResolvedCallableSymbol()) {
             is FirNamedFunctionSymbol -> data.add(symbol)
             is FirConstructorSymbol -> data.add(symbol)
+            null -> throw Exception("Expected ${functionCall.calleeReference} to be resolved, but it isn't.")
             else -> TODO("Are there any other possible cases?")
         }
 
         return data.withResult(calleeSig.returnType) {
             val args = getArgs().zip(calleeSig.formalArgs).map { (arg, formalArg) -> arg.withType(formalArg.viperType) }
-            data.addStatement(Stmt.MethodCall(calleeSig.name.mangled, args, listOf(this.resultVar.toLocalVar())))
+            data.addStatement(calleeSig.toMethodCall(args, this.resultVar))
         }
     }
 
     override fun visitImplicitInvokeCall(implicitInvokeCall: FirImplicitInvokeCall, data: StmtConversionContext): Exp {
         val args = getFunctionCallArguments(implicitInvokeCall).map(data::convert)
-        val retType = implicitInvokeCall.calleeReference.toResolvedCallableSymbol()!!.resolvedReturnType
+        val retType = implicitInvokeCall.toResolvedCallableSymbol()!!.resolvedReturnType
```

Comment:
maybe we can wrap calls like `toResolvedCallableSymbol()!!` in a private function in order to be null safe and make the code cleaner